jmh-benchmarks/src/main/java/org/apache/kafka/jmh/group_coordinator/ServerSideAssignorBenchmark.java

```diff
@@ -0,0 +1,235 @@
+
+        private final PartitionAssignor assignor;
+
+        AssignorType(PartitionAssignor assignor) {
+            this.assignor = assignor;
+        }
+
+        public PartitionAssignor assignor() {
+            return assignor;
+        }
+    }
+
+    /**
+     * The subscription pattern followed by the members of the group.
+     *
+     * A subscription model is considered homogenous if all the members of the group
+     * are subscribed to the same set of topics, it is heterogeneous otherwise.
+     */
+    public enum SubscriptionModel {
+        HOMOGENEOUS, HETEROGENEOUS
+    }
+
+    @Param({"1000", "10000"})
+    private int memberCount;
+
+    @Param({"10", "50"})
+    private int partitionsPerTopicCount;
+
+    @Param({"100", "1000"})
+    private int topicCount;
+
+    @Param({"true", "false"})
+    private boolean isRackAware;
+
+    @Param({"HOMOGENEOUS", "HETEROGENEOUS"})
+    private SubscriptionModel subscriptionModel;
+
+    @Param({"RANGE", "UNIFORM"})
+    private AssignorType assignorType;
+
+    @Param({"true", "false"})
+    private boolean simulateRebalanceTrigger;
+
+    private PartitionAssignor partitionAssignor;
+
+    private static final int numberOfRacks = 3;
+
+    private AssignmentSpec assignmentSpec;
+
+    private SubscribedTopicDescriber subscribedTopicDescriber;
+
+    @Setup(Level.Trial)
+    public void setup() {
+        Map<Uuid, TopicMetadata> topicMetadata = createTopicMetadata();
+        subscribedTopicDescriber = new SubscribedTopicMetadata(topicMetadata);
+
+        createAssignmentSpec(topicMetadata);
+
+        partitionAssignor = assignorType.assignor();
+
+        if (simulateRebalanceTrigger) {
+            simulateIncrementalRebalance(topicMetadata);
+        }
+    }
+
+    private Map<Uuid, TopicMetadata> createTopicMetadata() {
+        Map<Uuid, TopicMetadata> topicMetadata = new HashMap<>();
+        Map<Integer, Set<String>> partitionRacks = isRackAware ?
+            mkMapOfPartitionRacks(partitionsPerTopicCount) :
+            Collections.emptyMap();
+
+        for (int i = 1; i <= topicCount; i++) {
+            Uuid topicUuid = Uuid.randomUuid();
+            String topicName = "topic" + i;
+            topicMetadata.put(topicUuid, new TopicMetadata(
+                topicUuid,
+                topicName,
+                partitionsPerTopicCount,
+                partitionRacks
+            ));
+        }
+
+        return topicMetadata;
+    }
+
+    private void createAssignmentSpec(Map<Uuid, TopicMetadata> topicMetadata) {
+        Map<String, AssignmentMemberSpec> members = new TreeMap<>();
+        List<Uuid> allTopicIds = new ArrayList<>(topicMetadata.keySet());
+        int topicCounter = 0;
+
+        for (int i = 0; i < memberCount; i++) {
+            String memberName = "member" + i;
+            Optional<String> rackId = rackId(i);
+
+            // When subscriptions are homogenous, all members are assigned all topics.
+            List<Uuid> subscribedTopicIds;
```

Comment:
Should this be a Set? I think that there are cases where we call `assignmentMemberSpec.subscribedTopicIds().contains(topicId)`.